addrmgr/addrmanager.go

```diff
@@ -423,30 +422,36 @@ out:
 	for {
 		select {
 		case <-dumpAddressTicker.C:
-			a.savePeers()
+			err := a.savePeers()
+			if err != nil {
+				panic(errors.Wrap(err, "error saving peers"))
+			}
 
 		case <-a.quit:
 			break out
 		}
 	}
-	a.savePeers()
+	err := a.savePeers()
+	if err != nil {
+		panic(errors.Wrap(err, "error saving peers"))
+	}
 	a.wg.Done()
 	log.Trace("Address handler done")
 }
 
-// savePeers saves all the known addresses to a file so they can be read back
+// savePeers saves all the known addresses to the database so they can be read back
 // in at next run.
-func (a *AddrManager) savePeers() {
+func (a *AddrManager) savePeers() error {
 	a.mtx.Lock()
 	defer a.mtx.Unlock()
 
 	// First we make a serialisable datastructure so we can encode it to
 	// json.
-	sam := new(serializedAddrManager)
-	sam.Version = serialisationVersion
-	copy(sam.Key[:], a.key[:])
+	peersState := new(serializedPeersState)
```

Comment:
It could be nice if savePeers can be split into the serialization and the actual storing logic. Just the way loadPeers does (but please read the comment I wrote there before starting with this).